src/prefect/task_runners.py

```diff
@@ -323,6 +328,8 @@ async def _start(self, exit_stack: AsyncExitStack):
         """
         Start the process pool
         """
+        if self._max_workers is not None:
+            self._limiter = CapacityLimiter(self._max_workers)
```

Comment:
Could we set this to an `asyncnullcontext` when `_max_workers` is null so that the code in `_run_and_store_result` doesn't need a branch?